crates/clairvoyance/src/uniswap.rs

```diff
@@ -15,68 +15,129 @@ use bindings::uniswap_v3_factory::UniswapV3Factory;
+        self.inner.clone()
+    }
+    pub fn get_sqrt_price_x96(&self) -> ethers::types::U256 {
+        self.sqrt_price_x96
+    }
+    fn set_pool_tick(&mut self, tick: i32) {
+        self.tick = tick;
+    }
+    fn set_pool_liquidity(&mut self, liquidity: u128) {
+        self.liquidity = liquidity;
+    }
+    fn set_pool_sqrt_price_x96(&mut self, sqrt_price_x96: ethers::types::U256) {
+        self.sqrt_price_x96 = sqrt_price_x96;
+    }
+    /// Updates the pool tick and liquidity manually with a contract call.
+    pub async fn _update_pool_tick_and_price(&mut self) {
+        let slot_0 = self.inner.slot_0().call().await.unwrap();
+        self.set_pool_tick(slot_0.1);
+        self.set_pool_sqrt_price_x96(slot_0.0)
+    }
+    pub async fn _update_pool_liquidity(&mut self) {
+        self.set_pool_liquidity(self.inner.liquidity().call().await.unwrap());
+    }
     /// Public builder function that instantiates a `Pool`.
     pub async fn new(
         token_0: Token,
         token_1: Token,
         bp: u32,
         provider: Arc<Provider<Http>>,
-    ) -> Result<Self, ()> {
+    ) -> Result<Pool, ()> {
         match bp {
             1 | 5 | 30 | 100 => (),
             _ => return Err(()),
         }
-
-        let uniswap_v3_factory_address = "0x1F98431c8aD98523631AE4a59f267346ea31F984"
+        // Factory Address.
+        let uniswap_v3_factory_address = FACTORY
             .parse::<Address>()
             .unwrap();
 
+        // Factory contract object.
         let factory = UniswapV3Factory::new(uniswap_v3_factory_address, provider.clone());
-
+        // pool address from factory contract object
         let pool_address = factory
             .get_pool(token_0.address, token_1.address, bp * 100)
             .call()
             .await
             .unwrap();
-
-        Ok(Self {
+        // pool object
+        Ok(Pool {
             token_0,
             token_1,
+            bp,
             address: pool_address,
             factory,
             inner: IUniswapV3Pool::new(pool_address, provider.clone()),
+            tick: 0,
+            liquidity: 0,
+            sqrt_price_x96: ethers::types::U256::zero(),
         })
     }
 
     /// Monitor a pool for swap events and print to standard output.
     /// TODO: Make it print a `Swap` struct that implements fmt in a special way.
-    pub async fn monitor_pool(&self) {
-        let pool = &self.inner;
-        println!("Got Pool: {:#?}. Listening for events...", pool.address());
-        let tokens = (self.token_0.clone(), self.token_1.clone());
-        let swap_events = pool.swap_filter();
-        let pool_token_0 = pool.token_0().call().await.unwrap();
-        let mut swap_stream = swap_events.stream().await.unwrap();
+    pub async fn monitor_pool(&mut self) {
 
+        let pool_contract = self.get_pool_contract();
+        println!("Got Pool: {:#?}. Listening for events...", pool_contract.address());
+        let pool_tokens = self.get_pool_tokens();
```

Comment:
same things with getters and setters here. we can just expose the struct field publicly and access it here